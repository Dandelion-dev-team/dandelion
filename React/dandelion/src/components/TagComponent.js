import React, { useEffect, useState } from "react"
import "../styles/App.scss"

export default function TagComponent(props) {
  const [tags, setTag] = useState(0)

  useEffect(() => {
    //TESTED
    // Update the document title using the browser API
    fetch("http://localhost:3000/tag", {
      method: "GET",
      headers: new Headers({
        "Cache-Control": "no-cache, no-store, must-revalidate",
        Pragma: "no-cache",
        Expires: 0,
      }),
    })
      .then(response => response.json())
      .then(data => setTag(data))
  }, [])

  const deleteTag = index => {
    fetch("http://localhost:3000/tag/" + index, {
      method: "DELETE",
      headers: { "Content-Type": "application/json" },
    })
      .then(console.log("delete " + index))
      .then(window.location.reload(false))
  }

  const editTag = auth => {
    props.parentCallback(auth) 
  }

  return (
    <div className="recordTable">
      <table className="tableList">
        <thead>
          <tr>
            <th>ID</th>
            <th>Label</th>
            <th>Status</th>
          </tr>
        </thead>

        {tags
          ? tags.map(tag => (
              <tbody key={tags.id}>
                <td>{tag.id}</td>
                <td>{tag.label}</td>
                <td>{tag.status}</td>
                <td>
                  <div className="submit-btn">
                    <input
                      type="submit"
                      className="submitButton"
                      value="Edit"
                      onClick={() => {
                        editTag(tag)
                      }}
                    ></input>
                  </div>
                </td>
                <td>
                  <input
                    type="submit"
                    className="submitButton"
                    value="Delete"
                    onClick={() => {
                      deleteTag(tag.id)
                    }}
                  ></input>
                </td>
              </tbody>
            ))
          : null}
      </table>
    </div>
  )
}
