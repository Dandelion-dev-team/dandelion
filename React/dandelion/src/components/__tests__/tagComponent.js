import React from "react"
import renderer from "react-test-renderer"
import TagComponent from "../TagComponent"


describe("Tags", () => {
    let originalFetch;

    beforeEach(() => {
        originalFetch = global.fetch;
        global.fetch = jest.fn(() => Promise.resolve({
            json: () => Promise.resolve(
                {
                    "id": 1,
                    "label": "Caulifowler",
                    "status": "Active"
                },
                {
                    "id": 2,
                    "label": "Kohl Rabi",
                    "status": "Not Started"
                })
        }));
    });

    afterEach(() => {
        global.fetch = originalFetch;
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(<TagComponent />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

