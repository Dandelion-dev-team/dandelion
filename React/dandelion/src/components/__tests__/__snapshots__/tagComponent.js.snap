// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div
  className="authTable"
>
  <table
    className="tableList"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Label
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
  </table>
</div>
`;
